### ISSUE TITLE
Some AAMs not being detected

### ISSUE THREAD

**Issue by richard-jones** [CottageLabs/oacwellcome#78]
In a review of the AAMs detected by the app, some records are not getting an AAM detected when they should.

I have created an idealised CSV which contains PMCIDs for which there have been problems, and put it in the test resources directory: 

https://github.com/CottageLabs/oacwellcome/blob/develop/service/tests/resources/aam.csv

If you run this through the system, we find that most of them come back with aam "unknown", but they are all author manuscripts.  For example, see 

http://europepmc.org/articles/PMC4105573

Probably this is just a rule in OAG which is not detecting AAMs for that particular identifier, and just needs to be updated.

@emanuil-tolev - leaving this in your court for the moment, if you could take a look at what OAG is doing.  I will mail you separately in a moment the output spreadsheet that I generated from these ids, to save you some time in the initial diagnosis.

**Comment by richard-jones** [comment]
Oh also, remember that OAG will cache the responses, so we might need to clear the cache/archive of these identifiers to get them to start working.

**Comment by emanuil-tolev** [comment]
Er, this ID gives errors @richard-jones . Did you try opening the page?

I click on http://europepmc.org/articles/PMC4105573
and end up at http://europepmc.org/exception?2&1&0=UkpmcException .

None of the ID-s in that CSV (thanks for it) are opening, they all go to that error page. That could explain the problem....

**Comment by CecyMarden** [comment]
I have just been finding this exact issue when looking at the more recent spreadsheet I processed. I have emailed EBI about this. When I sent the file through to Richard I did check (i.e., actually open the page) that the articles found via the search were author manuscripts. HOWEVER, I did this on PMC (where the new articles I'm finding in the same situation also open fine), because you can only search by month on PMC. 

The reasoning doesn't really matter, what matters is that Europe PMC has something wrong with it here. EBI are investigating now (received an email as I wrote this). Will let you know when they have resolved the issue.

**Comment by richard-jones** [comment]
Yeah, they were fine when I checked a couple of them the other day, so this
must be a temporary fault at the EPMC end.

**Comment by emanuil-tolev** [comment]
> so this
> must be a temporary fault at the EPMC end.

Yeah, what I'm getting at is them erroring when OAG was looking at them, and then being OK when you looked at them. Anyway, we'll see when they're back up!

**Comment by CecyMarden** [comment]
I have just run the set of dodgy AAMs from the top of this thread, and all but two are still coming back as unknowns. I looked on Europe PMC at two of them, and I'm no longer getting the error message, so I'm not sure it can be a problem at Europe PMC. What do you see?

I do have some further information from EBI:

"To date we now know that our External Services Group modified some load balancer IPs, they then failed to pass these modifications onto the VM configurations. They are actually doing this now. Yesterday we were able to make a configuration change ourselves to get around the problem. Requests for content can go to different servers, and the user sessions would then 'stick' to particular servers according to the load balancers. So you can get situations were some content can be found but not all, it can appear quite strange at times (should you get such a problem).

Now why ES made this change I don't yet know, but we will be having words.
The use of the API should not have caused this problem , it's an in-house one."

I am still slightly unconvinced by this, but not sure how/whether to take this forward given that I can now see the articles and I don't know how I'd justify my feeling that it was the AAMs which had come back as unknown (previously) on our checks which were specifically having this issue. If you have any thoughts they're welcome.

**Comment by emanuil-tolev** [comment]
OK, I tried again with the 21 PMID-s that weren't working for AAM and it seems to all work now (I've made no code changes but I did delete all unknown EPMC results from howopenisit.org ). Here is my sheet on the test server: http://oacwellcome.ooz.cottagelabs.com/progress/66fb4a5bf9bf4379beb9ab14965b3d5a

AAM is all "true" now. It's mostly detected default copyright rules (so not-OA article) but this seems to meet compliance criteria - both Standard and Deluxe are met. Not sure if this is right, but I can't say I've got the truth table in my head :).

Anyhow, as far as AAM is concerned this seems to have been resolved when EPMC fixed that problem, so I'm closing this issue - feel free to open more issues if that spreadsheet looks wrong in other ways.
